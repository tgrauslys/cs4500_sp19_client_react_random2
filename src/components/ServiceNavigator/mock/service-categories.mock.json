[
  {
    "id": "123", "title": "Home Improvements",
    "services": [
      {"id": "1", "title": "Appliance Installation"},
      {"id": "2", "title": "Cabinetry"},
      {"id": "3", "title": "Door Installation"},
      {"id": "4", "title": "Concrete Removal"},
      {"id": "5", "title": "Dishwasher Installation"},
      {"id": "6", "title": "Finish Carpentry"},
      {"id": "7", "title": "Floor Polishing"},
      {"id": "8", "title": "Furniture Assembly"},
      {"id": "9", "title": "Generator Installation"},
      {"id": "10", "title": "Gardening"},
      {"id": "11", "title": "Home Organizing"},
      {"id": "12", "title": "Electrical and Wiring Repair"}
    ],
    "serviceProviders": [
      {
        "title": "Christina's Custom Cleaning, CCC",
        "rating": 4.4,
        "ratingScores": [
          93,
          24,
          12,
          10,
          4
        ],
        "yearsInBusiness": "7",
        "hires": 17,
        "latestReview": "It's so wonderful to have the burden of cleaning - especially with an injury - removed. Christina is excellent with communication and I've been very pleased with the quality of work.",
        "price": "$75",
        "introduction": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum",
        "backgroundChecked": true,
        "employees": 2,
        "paymentMethods": [
          "Cash", "Venmo"
        ],
        "faqs": [
          {
            "question": "What should the customer know about your pricing (e.g., discounts, fees)?",
            "answer": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum"
          },
          {
            "question": "What education and/or training do you have that relates to your work?",
            "answer": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum"
          }
        ],
        "reviews": [
          {
            "id": 1,
            "who": "Kristine M.",
            "stars": 5,
            "review":"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
            "date": "Apr 13, 2017",
            "reply": "thank you very much!! see you next weekend!!"
          },
          {
            "id": 2,
            "who": "Eric G.",
            "stars": 5,
            "review":"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
            "date": "Apr 19, 2017",
            "reply": "Thank you so much Eric! We loved cleaning your home! "
          }
        ]
      },
      {
        "title": "Perfect Shine Cleaner",
        "rating": 5,
        "reviews": 1,
        "yearsInBusiness": "1",
        "hires": 3,
        "latestReview": "Ana came is and did an amazing job cleaning before a family party. She was available on short notice and exceeded my expectations. Thanks Ana!",
        "price": "$90"
      },
      {
        "title": "Mendes cleaning services",
        "rating": 4.8,
        "reviews": 35,
        "yearsInBusiness": "5",
        "hires": 23,
        "latestReview": "Wonderful job! House was quite a mess and they did a great job! Scheduling them for regular cleanings! High recommend!!!",
        "price": "$90"
      }
    ],
    "questions": [
      {
        "id": "1",
        "question": "Number of bedrooms",
        "questionType": "MULTIPLE-CHOICE",
        "choices": [
          {"id": "1", "title": "1 bedroom"},
          {"id": "2", "title": "2 bedroom"},
          {"id": "3", "title": "3 bedroom"},
          {"id": "4", "title": "4 bedroom"},
          {"id": "5", "title": "5 bedroom"},
          {"id": "6", "title": "6 bedroom"}
        ]},
      {"id": "2", "question": "Number of bathrooms", "questionType": "MULTIPLE-CHOICE",
        "choices": [
          {"id": "1", "title": "1 bathroom"},
          {"id": "2", "title": "2 bathroom"},
          {"id": "3", "title": "3 bathroom"},
          {"id": "4", "title": "4 bathroom"},
          {"id": "5", "title": "5 bathroom"},
          {"id": "6", "title": "6 bathroom"}
        ]},
      {"id": "3", "question": "Cleaning type", "questionType": "MULTIPLE-CHOICE",
        "choices": [
          {"id": "1", "title": "Standard cleaning"},
          {"id": "2", "title": "Deep cleaning"},
          {"id": "3", "title": "Move out cleaning"}
        ]},
      {"id": "4", "question": "Extra services", "questionType": "MULTIPLE-ANSWER",
        "choices": [
          {"id": "1", "title": "Fridge cleaning"},
          {"id": "2", "title": "Oven cleaning"},
          {"id": "3", "title": "Window cleaning"},
          {"id": "4", "title": "Laundry"}
        ]},
      {"id": "5", "question": "Frequency", "questionType": "MULTIPLE-CHOICE",
        "choices": [
          {"id": "1", "title": "Just once"},
          {"id": "2", "title": "Every week"},
          {"id": "3", "title": "Every 2 weeks"},
          {"id": "4", "title": "Once a month"}
        ]}
    ]
  },
  {
    "id": "345", "title": "Pets",
    "serviceProviders": [],
    "questions": [],
    "services": [
    {"id": "1", "title": "Cat Grooming"},
    {"id": "2", "title": "Dog Training"},
    {"id": "3", "title": "Pet Boarding"},
    {"id": "4", "title": "Pet Sitting"},
    {"id": "5", "title": "Dog Walking"},
    {
      "id": "6",
      "title": "Dog Grooming",
      "serviceProviders": [
        {
          "title": "Lucky Dogs Daycare & Pet Spa",
          "rating": 4.4,
          "reviews": 21,
          "yearsInBusiness": "7",
          "hires": 17,
          "latestReview": "This weekend we  took her for a two-hour play time followed by ...",
          "price": "$35/Dog"
        },
        {
          "title": "Zenful Pawz LLC",
          "rating": 5,
          "reviews": 1,
          "yearsInBusiness": "1",
          "hires": 3,
          "latestReview": "! Sheâ€™s the best! I know my little Betty is in great care. We have been to many groomers & this is the best yet! We will continue to visit Izzy!",
          "price": "contact for price"
        },
        {
          "title": "Fluff & Buff Boutique",
          "rating": 4.8,
          "reviews": 35,
          "yearsInBusiness": "5",
          "hires": 23,
          "latestReview": "Kate really knows how to handle dogs. Her love of animals shines and she takes great pride in her work. My dog always looks great after her care.",
          "price": "contact for price"
        }
      ],
      "questions": [
        {"id": "1", "question": "Dog size",
          "questionType": "MULTIPLE-CHOICE",
          "choices": [
            {"id": "1", "title": "Extra small (1 - 15 lbs)"},
            {"id": "2", "title": "Small (16 - 25 lbs)"},
            {"id": "3", "title": "Medium (26 - 40 lbs)"},
            {"id": "4", "title": "Large (41 - 70 lbs)"},
            {"id": "5", "title": "Extra large (70+ lbs)"}
          ]},
        {"id": "2", "question": "Dog coat type",
          "questionType": "MULTIPLE-CHOICE",
          "choices": [
            {"id": "1", "title": "Smooth, skin-like coat"},
            {"id": "2", "title": "Wavy or curly coat"},
            {"id": "3", "title": "Silky coat with fine hair"},
            {"id": "4", "title": "Wiry, bristly coat"},
            {"id": "5", "title": "Thick, fluffy double-coat"}
          ]}
      ]},
    {"id": "7", "title": "Dog Daycare"},
    {"id": "8", "title": "Aquarium Services"},
    {"id": "9", "title": "Animal Training (Non-canine)"},
    {"id": "10", "title": "Behavior Modification (Non-canine)"}
  ]},
  {
    "id": "234", "title": "Wellness",
    "serviceProviders": [],
    "questions": [],
    "services": []},
  {
    "id": "456", "title": "Business",
    "serviceProviders": [],
    "questions": [],
    "services": []},
  {
    "id": "567", "title": "Events",
    "serviceProviders": [],
    "questions": [],
    "services": []},
  {
    "id": "678", "title": "Crafts",
    "serviceProviders": [],
    "questions": [],
    "services": []},
  {
    "id": "789", "title": "Design and Web",
    "serviceProviders": [],
    "questions": [],
    "services": []},
  {
    "id": "890", "title": "Legal",
    "serviceProviders": [],
    "questions": [],
    "services": []},
  {
    "id": "987", "title": "Personal",
    "serviceProviders": [],
    "questions": [],
    "services": []},
  {
    "id": "876", "title": "Photography",
    "serviceProviders": [],
    "questions": [],
    "services": []}
]