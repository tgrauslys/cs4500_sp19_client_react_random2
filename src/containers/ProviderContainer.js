import React from 'react'
import Provider from "../components/Provider/Provider";

class ProviderContainer extends React.Component {

    constructor(props) {
        super(props);
        this.userService = props.UserService;
        this.reviewService = props.ReviewService;
        this.faqAnswerService = props.FAQAnswerService;
        this.providerId = this.props.id;
        this.state = {
            provider : {},
            reviews : [],
            ratingScores : [],
            FAQAnswers : [],

            reviewDescription : "",
            reviewRating : "",
            reviewUserFrom : "",

        }
    }


    componentDidMount() {
        this.userService.findUserById(this.providerId).then(provider => this.setState({provider: provider})).catch();
        this.reviewService.findReviewsForUser(this.providerId).then(reviews => this.setState({reviews: reviews})).catch();
        this.reviewService.findRatingScores(this.providerId).then(ratingScores => this.setState({ratingScores : ratingScores})).catch();
        this.faqAnswerService.findFAQsByUserId(this.providerId).then(FAQAnswers => this.setState({FAQAnswers : FAQAnswers})).catch();
    }

    handleChange(e) {
        this.setState({[e.target.name]: e.target.value})
    }

    submitReview(e) {
        let review = {}
        review["description"] = this.state.reviewDescription;
        review["rating"] = this.state.reviewRating;
        this.userService.profile().then(user => {
            this.reviewService.createReview(review).then(reviewresponse => {
                console.log(reviewresponse);
                this.reviewService.linkUserFrom(reviewresponse.id, user.id).then( linkresponse => {
                        console.log(reviewresponse)
                        this.reviewService.linkUserTo(reviewresponse.id, this.providerId)
                }

                );
            })});
    }


    render () {
        return <Provider
                    provider={this.state.provider}
                    reviews={this.state.reviews}
                    ratings={this.state.ratingScores}
                    FAQAnswers={this.state.FAQAnswers}
                    handleChange={this.handleChange.bind(this)}
                    reviewDescription={this.state.reviewDescription}
                    reviewRating={this.state.reviewRating}
                    submitReview={this.submitReview.bind(this)}/>
    }
}
export default ProviderContainer