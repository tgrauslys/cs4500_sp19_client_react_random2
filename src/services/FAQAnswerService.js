export default class FAQAnswerService {
    static instance = null;
    static getInstance() {
        if(FAQAnswerService.instance === null) {
            FAQAnswerService.instance = new FAQAnswerService()
        }
        return this.instance
    }
    findFAQAnswerById = id =>
        fetch(`http://localhost:8080/api/faq-answers/${id}`)
            .then(response => response.json())
    findAllFAQAnswers = () =>
        fetch("http://localhost:8080/api/faq-answers")
            .then(response => response.json())
}