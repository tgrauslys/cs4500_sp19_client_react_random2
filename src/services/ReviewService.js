export default class ReviewService {
    static instance = null;

    static getInstance() {
        if (ReviewService.instance === null) {
            ReviewService.instance = new ReviewService()
        }
        return this.instance
    }


    findReviewsForUser = id =>
        fetch(`${process.env.REACT_APP_MIDDLE_TIER_URL}/api/userto/${id}/reviews`)
            .then(response => response.json())

    findRatingScores = id =>
        fetch(`${process.env.REACT_APP_MIDDLE_TIER_URL}/api/ratingscores/${id}`).then(response => response.json())


    createReview = review =>
        fetch(`${process.env.REACT_APP_MIDDLE_TIER_URL}/api/review`, {
            method: 'post',
            headers: {
                "Content-Type": "application/json",
                // "Content-Type": "application/x-www-form-urlencoded",
            },
            body: JSON.stringify(review), // body data type must match "Content-Type" header
        }).then(response => response.json());

    linkUserTo = (reviewId, userId) => fetch(`${process.env.REACT_APP_MIDDLE_TIER_URL}/api/review/${reviewId}/userto/${userId}`,
        {method: 'post'});

    linkUserFrom = (reviewId, userId) => fetch(`${process.env.REACT_APP_MIDDLE_TIER_URL}/api/review/${reviewId}/userfrom/${userId}`,
        {method: 'post'});
}


