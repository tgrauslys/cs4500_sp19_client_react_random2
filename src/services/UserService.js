export default class UserService {
    static instance = null;
    static getInstance() {
        if(UserService.instance === null) {
            UserService.instance = new UserService()
        }
        return this.instance
    }
    findUserById = userId =>
        fetch(`http://localhost:8080/api/users/${userId}`)
            .then(response => response.json())
    findAllUsers = () =>
        fetch("http://localhost:8080/api/users")
            .then(response => response.json())
}
