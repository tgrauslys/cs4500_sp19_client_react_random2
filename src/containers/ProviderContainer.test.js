import React from 'react'

import ReviewService from '../services/ReviewService'
import UserService from '../services/UserService'
import FAQAnswerService from '../services/FAQAnswerService'

import '../services/UserService.mock'


import TestRenderer from 'react-test-renderer'
import ProviderContainer from "./ProviderContainer";


const reviewService = ReviewService.getInstance();
const userService = UserService.getInstance();
const faqAnswerService = FAQAnswerService.getInstance();


test('Render profile correctly', async () => {
    const testRenderer = TestRenderer.create(
            <ProviderContainer
                id = {1}
                ReviewService = {reviewService}
                UserService =  {userService}
                FAQAnswerService = {faqAnswerService}/>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();

});