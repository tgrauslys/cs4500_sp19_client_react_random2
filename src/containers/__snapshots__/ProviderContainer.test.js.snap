// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render profile correctly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-3 text-right"
    >
      <a
        href="#"
      >
        Sign up
      </a>
    </div>
    <div
      className="col-1"
    >
      <a
        href="#"
      >
        Log in
      </a>
    </div>
  </div>
  <br />
  <br />
  <ul
    className="nav nav-pills"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="#about"
      >
        About
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="#reviews"
      >
        Reviews
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="#faqs"
      >
        FAQs
      </a>
    </li>
  </ul>
  <div
    className="row"
  >
    <div
      className="col-2"
    >
      <a
        name="about"
      />
      <img
        src="https://picsum.photos/150/150"
      />
    </div>
    <div
      className="col-10"
    >
      <h3 />
      <i
        className="fa fa-star cs4500-yellow"
      />
      <i
        className="fa fa-star cs4500-yellow"
      />
      <i
        className="fa fa-star cs4500-yellow"
      />
      <i
        className="fa fa-star cs4500-yellow"
      />
      <i
        className="fa fa-star cs4500-yellow"
      />
      (
      0
      )
    </div>
  </div>
  <div>
    <br />
    <p />
  </div>
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <h4>
        Overview
      </h4>
      <i
        className="fa fa-trophy"
      />
        Hired 
       times
      <br />
      <i
        className="fa fa-lock"
      />
       
      
      <br />
      <i
        className="fa fa-users"
      />
       
       Employees
      <br />
      <i
        className="fa fa-briefcase"
      />
       
    </div>
    <div
      className="col-6"
    >
      <h4>
        Payment methods
      </h4>
      <i
        className="fa fa-usd"
      />
    </div>
  </div>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-8"
    >
      <a
        name="reviews"
      />
      <h3>
        Reviews
      </h3>
    </div>
    <hr />
    <div
      className="col-4"
    >
      <select
        className="form-control float-right"
      >
        <option>
          Recommended
        </option>
        <option>
          Highest rated
        </option>
        <option>
          Lowest rated
        </option>
        <option>
          Newest first
        </option>
        <option>
          Oldest first
        </option>
      </select>
      <br />
      <br />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      <h4>
        Rating: undefined
      </h4>
      <i
        className="fa fa-star cs4500-yellow wd-font-size-2-em"
      />
      <i
        className="fa fa-star cs4500-yellow wd-font-size-2-em"
      />
      <i
        className="fa fa-star cs4500-yellow wd-font-size-2-em"
      />
      <i
        className="fa fa-star cs4500-yellow wd-font-size-2-em"
      />
      <i
        className="fa fa-star cs4500-yellow wd-font-size-2-em"
      />
      <br />
      0
       reviews
    </div>
    <div
      className="col-8"
    />
  </div>
  <br />
  <div>
    <div
      className="center"
    >
      <nav
        aria-label="..."
      >
        <ul
          className="pagination"
        >
          <li
            className="page-item disabled"
          >
            <span
              className="page-link"
            >
              Previous
            </span>
          </li>
          <li
            className="page-item"
          >
            <a
              className="page-link"
              href="#"
            >
              1
            </a>
          </li>
          <li
            aria-current="page"
            className="page-item active"
          >
            <span
              className="page-link"
            >
              2
              <span
                className="sr-only"
              >
                (current)
              </span>
            </span>
          </li>
          <li
            className="page-item"
          >
            <a
              className="page-link"
              href="#"
            >
              3
            </a>
          </li>
          <li
            className="page-item"
          >
            <a
              className="page-link"
              href="#"
            >
              Next
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
  <div>
    <input
      className="review-fld"
      name="reviewDescription"
      onChange={
        Array [
          [Function],
        ]
      }
      value=""
    />
    <select
      className="star-fld"
      name="reviewRating"
      onChange={
        Array [
          [Function],
        ]
      }
      value=""
    >
      <option
        value={1}
      >
        1
      </option>
      <option
        value={2}
      >
        2
      </option>
      <option
        value={3}
      >
        3
      </option>
      <option
        value={4}
      >
        4
      </option>
      <option
        value={5}
      >
        5
      </option>
    </select>
    <button
      className="submit-review-btn"
      onClick={[Function]}
    >
      Submit Review
    </button>
  </div>
  <div>
    <hr />
    <a
      name="faqs"
    />
    <h3>
      FAQs
    </h3>
  </div>
</div>
`;
